perllib/FixMyStreet/Cobrand/Bexley/Garden.pm

```diff
@@ -30,14 +35,27 @@ sub garden_current_subscription {
     my $uprn = $self->{c}->stash->{property}{uprn};
     return undef unless $uprn;
 
-    my $is_free = $self->agile->IsAddressFree($uprn);
-    return undef if $is_free->{IsFree} eq 'True';
+    my $results = $self->agile->CustomerSearch($uprn);
+    ::Dwarn($results);
+    return undef unless $results && $results->{Customers};
+    my $customer = $results->{Customers}[0];
+    return undef unless $customer && $customer->{ServiceContracts};
+    my $contract = $results->{ServiceContracts}[0];
+    return unless $contract;
 
     # Agile says there is a subscription; now get service data from
     # Whitespace
     my $services = $self->{c}->stash->{services};
-    map { my $srv = $services->{$_}; return $srv if $srv }
-        @{ $self->garden_service_ids };
+    for my $id (@{ $self->garden_service_ids }) {
```

Comment:
Not sure if it's something that needs solving right now, but I am not sure of the best way to handle the Whitespace/Agile split; e.g. if we have Agile data for the contract but not Whitespace. Perhaps how we're doing it now will suffice 🤔 

In my cancellation PR I have extended the `agile_only` bit below to include a couple more fields like `customer_external_ref` and `end_date` (I *think* this can be used as the 'DueDate' required by the cancel call, but I'll need to check that with Luis). I guess I should add those to the Whitespace $srv data here as well.